pkg/sql/conn_executor.go

```diff
@@ -1200,6 +1200,12 @@ func (ex *connExecutor) run(
 				res = ex.clientComm.CreateErrorResult(pos)
 				break
 			}
+
+			if portal.Stmt.AST == nil {
+				res = ex.clientComm.CreateEmptyQueryResult(pos)
+				break
+			}
+
 			if log.ExpensiveLogEnabled(ex.Ctx(), 2) {
 				log.VEventf(ex.Ctx(), 2, "portal resolved to: %s", portal.Stmt.AST.String())
```

Comment:
tip: if you were to pass `portal.Stmt.AST` by reference without calling `String()` upfront, you could drop `ExpensiveLogEnabled`: the log event formatter would only call `String()` if the event was enabled.

Reply:
Thanks - just want to keep this a backport for now though